lib/src/ruby/resolution.cc

```diff
@@ -183,9 +183,14 @@ namespace facter { namespace ruby {
     {
         auto const& ruby = api::instance();
 
+        int64_t val = ruby.rb_num2ll(value);
+        if (val < 0) {
```

Comment:
Before this change would we silently cast negative numbers to huge unsigned numbers?